the-collab-lab/tcl-30-whats-near-me
# [TCL30-10] Render List Nearby API

## Description

This code allows the user to see Wikipedia entries, in a table and/or list, of places nearby. 

To accomplish this, we added the file `data` with two `.json` files, `paris.json` and `zero.json`. These will be removed once it has been merged with [TCL30-9](https://the-collab-lab.atlassian.net/browse/TCL30-9). 

We created a custom hook, `usePages.js`. It uses the latitude and longitude coordinates for the center of the map, `centerMap`, that are declared in `List.js`. 

We also created the file `fixturesnearby.js`. This will process and return the location data from one of the two `.json` files.

One thing we learned is that the coordinates for latitude and longitude need to be consistently the exact same number among all of the components to properly render. This would be important to remember in case any attempts at rounding those numbers is suggested.


## Related Issue

Closes [TCL30-10](https://the-collab-lab.atlassian.net/browse/TCL30-10)


## Acceptance Criteria

- When the app starts, a call should be made to the proxy endpoint using the latitude and longitude at the center of the map (until the API calling code is merged, just use your fixture)

- The returned data should be used to render a table or list of information (sorted by distance) on the /list route that includes the following:

  - Distance (in km)

  - Thumbnail (if present, scaled down to 24px x 24px using object-fit: cover)

  - Name

  - Description

  - A link to view the place on Wikipedia

- Links out to Wikipedia should use the `pageid` property that is returned by the API, using the following format: `https://en.wikipedia.org/?curid=<pageid>` (e.g., https://en.wikipedia.org/?curid=60401407)

**We are not using the user’s device location at this point.** 


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|    | :link: Update dependencies |
|     | :scroll: Docs              |


## Updates

 List View - BEFORE   |  List View - AFTER
:----------------------------------:|:-------------------------------------:
![image](https://user-images.githubusercontent.com/8689897/126203134-3974c014-8493-4a00-bdb7-d0abdd402d68.png) | ![image](https://user-images.githubusercontent.com/8689897/126841205-28cf9240-8c1f-45df-b3d1-cd83a985b033.png)



## Testing Steps / QA Criteria

1. navigate to the branch `ac-jw-render-list-nearby`
2. run `npm install`
3. run `npm start`
4. navigate to the `List` view
5. if you would like to see another location in the `List` view, do the following:
  - In your code editor, open `src/components/List.js`
  - between lines 7 and 10, you will have two sets of `lat` and `lng` to choose from. You can comment out the currently active set of coordinates, and uncomment the other set of coordinates to make them active. 
  
<img width="572" alt="TCL30-10_TestingIMG" src="https://user-images.githubusercontent.com/38255956/126073810-92272c79-6120-4cc9-a921-63a86478996b.png">

  - if you do not have autosave, save the file
  - on `localhost:3000`, navigate to the "/list" route, and see the results.
